packages/oracle/src/worker/worker.ts

```diff
@@ -144,15 +146,26 @@ export class OracleWorker implements Worker {
       logger.debug(
         `For pool ${poolMock.id} token0 is ${token0.id} (${token0.decimals}), token1 is ${token1.id}, (${token1.decimals})`
       );
-      const priceFp18 = priceToPriceFp18(token0Price, token0.decimals, token1.decimals);
       const sqrtPriceX96 = priceToSqrtPriceX96(token0Price, token0.decimals, token1.decimals);
+
+      let priceFp;
+      let fpNumber;
+      if (priceDenominator.eq(BigNumber.from(10).pow(27))) {
+        priceFp = priceToPriceFp18(token0Price, token0.decimals, token1.decimals);
+        fpNumber = 27;
+      } else {
```

Comment:
I think, it would be better to throw an error in case if fsNumber is neither 27 nor 18.